tutorials/Inference-TensorFlow-Bert-Model-for-High-Performance-in-ONNX-Runtime.ipynb

```diff
@@ -0,0 +1,211 @@
+   "source": [
+    "Run the `bert_model_optimization.py` with `--framework tensorflow` option to optimize the converted model. Other notable options to use are \n",
+    "- `--gpu_only`: allow half-precision float for better performance.\n",
+    "- `--input_int32`: Use int32 tensors instead of default int64 as input to avoid un-necessary Cast nodes and get better performance.\n",
+    "- `--float16`: Use float16 tensors instead of default float32 as input to enable half precision floats. Recommended for NVidia GPU with Tensor Core like V100 and T4. For older GPUs, float32 is likely faster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Below are three examples to run bert_model_optimization.py. Choose one according to your needs and adjust --input\n",
+    "# --output path names as necessary.\n",
+    "\n",
+    "# For CPU\n",
+    "!python bert_op_scripts/bert_model_optimization.py --input <bert.onnx> --output <bert_cpu.onnx> --framework tensorflow\n",
+    "\n",
+    "# # For inferences under NVidia GPU with Tensor Core like V100 and T4\n",
+    "# !python bert_op_scripts/bert_model_optimization.py --input <bert.onnx> --output <bert_gpu_fp16.onnx> --framework tensorflow --gpu_only –float16\n",
+    "\n",
+    "# # For inferences under other NVidia GPUs except V100 and T4\n",
+    "# !python bert_op_scripts/bert_model_optimization.py --input <bert.onnx> --output <bert_gpu_fp32.onnx> --framework tensorflow --gpu_only\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3 - Inference the Optimized Model with ONNX Runtime\n",
+    "\n",
+    "#### Install ONNX Runtime\n",
+    "Install the latest ONNX Runtime if you haven't done so already. \n",
+    "\n",
+    "Install one `onnxruntime` python build. Choose to install `onnxruntime` to use CPU features, or `onnxruntime-gpu` to enjoy GPU execution providers such as CUDA. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Install ONNX Runtime for CPU\n",
+    "!{sys.executable} -m pip install -U onnxruntime\n",
+    "\n",
+    "## Alternatively, install onnxruntime for GPU\n",
+    "# !{sys.executable} -m pip install -U onnxruntime-gpu"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we're ready to use ONNX Runtime to do inference on the optimized model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import onnxruntime as rt  \n",
+    "import numpy as np\n",
+    "import time\n",
+    "\n",
+    "sess_options = rt.SessionOptions()\n",
+    "\n",
+    "# Set graph optimization level to ORT_ENABLE_EXTENDED to enable bert optimization.\n",
+    "sess_options.graph_optimization_level = rt.GraphOptimizationLevel.ORT_ENABLE_EXTENDED\n",
```

Comment:
same comment as in PT tutorial: shouldn't this be ALL

Reply:
mark as 'resolved' as duplicated